pylintrc

```diff
@@ -353,6 +353,13 @@ method-rgx=[a-z_][a-z0-9_]{2,}$
 # Naming hint for method names
 method-name-hint=[a-z_][a-z0-9_]{2,}$
 
+# Naming style used for type variables. Default is PascalCase.
+typevar-naming-style=PascalCase
```

Comment:
I believe this was the most recent discussion we had: Should we add a `typevar-naming-style` add all or only have a `typevar-rgx` option?

The arguments so far
* If we say the default is `PascalCase`, it will be difficult to explain why `PathType` is invalid and `PathT` valid when both are PascalCase.
* It's unclear what the regex patterns for the other styles should be and how they should interact with the `_co` and `_contra` suffixes.
* Having only the the rgx option (with default) would still allow for customizability if wanted.
* What should be done in similar cases, e.g. for ParamSpec names or Type aliases?

Changing it should be possible if we modify `_create_naming_rules` and `_create_naming_options` to allow for rgx only options.

Reply:
I was hesitant to do this at first, but turned out to be relatively easy to implement. I have pushed an iteration which does this to this branch. This basically supersedes #5221. I'm also fine with pushing it to that branch to keep the discussion in one place, but perhaps a relatively clean slate is helpful here.